text/0000-route-model-name-property.md

```diff
@@ -0,0 +1,149 @@
+
+> Would the acceptance of this proposal mean the Ember guides must be
+re-organized or altered? Does it change how Ember is taught to new users
+at any level?
+
+We would continue to mention the application of the model hook in the guides as we currently do:
+
+> [_"The route will then set the return value from the model hook as the model property of the controller."_](https://guides.emberjs.com/release/routing/specifying-a-routes-model/)
+
+We could add an extra sentence covering this feature:
+
+You can also set the return value from the model hook to be any custom property name on the controller by using `modelName` property in the route.
+```
+app/routes/posts.js
+import Route from '@ember/routing/route';
+
+export default Route.extend({
+  model() {
+    return this.store.findAll('post');
+  },
+  modelName: articles
+});
+```
+```
+app/templates/posts.hbs
+<h1>Articles</h1>
+{{#each this.articles as |article|}}
+  <p>{{articles.body}}</p>
+{{/each}}
+```
+> How should this feature be introduced and taught to existing Ember
+users?
+
+We can let developers know that instead of doing:
+```
+app/routes/posts.js
+import Route from '@ember/routing/route';
+
+export default Route.extend({
+  model() {
+    return this.store.findAll('post');
+  },
+  setupController(model, controller){
+    controller.set('customModelName', model);
+  }
+});
+```
+They can simply use  the `modelName` property on the route, if they want to 😀.
+
+## Drawbacks
+
+> Why should we *not* do this? Please consider the impact on teaching Ember,
+on the integration of this feature with other existing and planned features,
+on the impact of the API churn on existing apps, etc.
+
+Increases the size of the API.
+
+> There are tradeoffs to choosing any path, please attempt to identify them here.
+
+## Alternatives
+
+> What other designs have been considered? What is the impact of not doing this?
+
+We currently have a way of doing this:
+```
+app/routes/posts.js
+import Route from '@ember/routing/route';
+
+export default Route.extend({
+  model() {
+    return this.store.findAll('post');
+  },
+  setupController(model, controller){
+    controller.set('customModelName', model);
+  }
+});
+```
+
+I thought about using decorators, but I'm unsure if that would even be feasible.
+
+I opened this PR to get feedback and hear other alternatives.
+
+I don't consider this to be essential feature, the RFC is born out of my personal frustration with Ember. In large code bases I often find that the return from the model hook is not the resource represented by the file name. I make the assumption that the `model` in the controller is what the convention suggests it would be, and it's a reoccuring 'gotcha' moment, that costs me time when I have to debug and realise that the `model` is different to the expectation I have from Ember's conventions.
```

Comment:
```suggestion
I don't consider this to be an essential feature, the RFC is born out of my personal frustration with Ember. In large code bases I often find that the return from the model hook is not the resource represented by the file name. I make the assumption that the `model` in the controller is what the convention suggests it would be, and it's a reoccuring 'gotcha' moment, that costs me time when I have to debug and realise that the `model` is different to the expectation I have from Ember's conventions.
```